components/StyledDatePickerController/StyledDatePickerController.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import { DayPickerSingleDateController } from 'react-dates';
+import { Icon } from 'semantic-ui-react';
+import 'react-dates/initialize';
+import 'react-dates/lib/css/_datepicker.css';
+import './styles.less';
+
+export const StyledDatePickerController = 
+  ({ highlightedDates, onDateChange, disablePrev, disableNext, date, ...passthroughProps }) => {
+    const [selectedDate, setSelectedDate] = useState(date);
+
+    function onDateSelect(newDate) {
+      setSelectedDate(newDate);
+      if(onDateChange){
+        onDateChange(newDate);
+      }
+    }
+
+    function isDayHighlighted(day) {
+      return highlightedDates && highlightedDates.some(highlightedDate => day.isSame(highlightedDate, 'day'));
+    }
+
+    function isDayBlocked(day) {
+      return highlightedDates && !highlightedDates.includes(day.format('YYYY-MM-DD'));
+    }
+
+    return (
+      <div className="singleDatePickerWrapper">
+        <DayPickerSingleDateController
+          {...passthroughProps}
+          noBorder
+          hideKeyboardShortcutsPanel
+          numberOfMonths={1}
+          date={selectedDate}
+          onDateChange={onDateSelect}
```

Comment:
Not blocking, but I'd rather see `onDateChange={onDateChange}` here, instead of nesting it inside `onDateSelect()` but maybe it has to be that way.

Reply:
It has to be this way, as we need to close the calendar after selection.